usbsdmux/__main__.py

```diff
@@ -22,24 +22,26 @@
 import errno
 import sys
 
-from .usbsdmux import UsbSdMux
+from .usbsdmux import autoselect_driver, UnknownUsbSdMuxRevisionException
 
 
 def main():
     parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter)
 
     parser.add_argument("sg", metavar="SG", help="/dev/sg* to use")
-    parser.add_argument(
-        "mode",
-        help="Action:\n"
-        "get - return selected mode\n"
-        "dut - set to dut mode\n"
-        "client - set to dut mode (alias for dut)\n"
-        "host - set to host mode\n"
-        "off - set to off mode",
-        choices=["get", "dut", "client", "host", "off"],
-        type=str.lower,
-    )
+    subparsers = parser.add_subparsers(help="Supply one of the following commands to interact with the device")
+    subparsers.required = True
+    subparsers.dest = "mode"
+
+    subparsers.add_parser("get", help="Read the current state of the USB-SD-Mux")
+    subparsers.add_parser("dut", help="Switch to the DUT")
+    subparsers.add_parser("client", help="Switch to the DUT")
+    subparsers.add_parser("host", help="Switch to the host")
+    subparsers.add_parser("off", help="Disconnect from host and DUT")
+
+    parser_gpio = subparsers.add_parser("gpio", help="Manipulate a GPIO")
+    parser_gpio.add_argument("gpio", help="The GPIO to change", choices=[0, 1], type=int)
+    parser_gpio.add_argument("action", help="What to do with the GPIO", choices=["low", "0", "high", "1", "get"])
```

Comment:
Our _GPIOs_ can not be used as inputs, but I think the name is okay nevertheless.
I've been thinking about alternatives but did not come up with a good one yet:

- `output` - too generic
- `gpo` - not as descriptive as gpio, which is a well known term

Reply:
What about describing it like this:
```
    parser_gpio = subparsers.add_parser("gpio", help="Manipulate a GPIO (open drain output only)")
```

This way this would be reflected in the ``--help``:
```
./venv/bin/usbsdmux --help
usage: usbsdmux [-h] SG {get,dut,client,host,off,gpio} ...

positional arguments:
  SG                    /dev/sg* to use
  {get,dut,client,host,off,gpio}
                        Supply one of the following commands to interact with the device
    get                 Read the current state of the USB-SD-Mux
    dut                 Switch to the DUT
    client              Switch to the DUT
    host                Switch to the host
    off                 Disconnect from host and DUT
    gpio                Manipulate a GPIO (open drain output only)

optional arguments:
  -h, --help            show this help message and exit
```

Reply:
Sounds good. That should avoid confusion for users that find the sub-command and its `get` option by exploring the tool rather than via the manual.